lib/travis/addons/config.rb

```diff
@@ -17,24 +17,43 @@ def initialize(payload, secure_key = nil)
       end
 
       def enabled?(key)
-        return false unless notifications.respond_to?(:has_key?)
-        return !!notifications[key] if notifications.has_key?(key) # TODO this seems inconsistent. what if email: { disabled: true }
-        [:disabled, :disable].each { |key| return !notifications[key] if notifications.has_key?(key) } # TODO deprecate disabled and disable
-        true
+        notifications.any? do |notifier|
+          if !notifier.respond_to?(:has_key?)
+            false
+          elsif notifier.has_key?(key)
+            !!notifier[key]
+          elsif notifier.has_key?(:disabled)
+            !notifier[:disabled]
+          elsif notifier.has_key?(:disable)
+            !notifier[:disable]
+          else
+            true
+          end
+        end
       end
 
       def send_on?(type, event)
-        Notify.new(build, notifications).on?(type, event)
+        notifications.any? do |notifier|
```

Comment:
I think this needs extra thoughts. Each notification needs to be considered whether or not it should fire, not `#any?`.

Reply:
Actually, it _is_ considered, by passing each configuration down to a `Notify` object, which will return `true` if the configuration defines a `send`-able notification.